modules/ROOT/pages/_partials/data-sync/server-offline-and-conflict.adoc

```diff
@@ -108,164 +64,104 @@ type Mutation {
 }
 ----
 
-. Implement the resolver.
-
-Conflicts can be resolved either on the client or the server. Depending on the strategy used, the resolver implementation will differ.
-See the sections below for the individual implementations.
-
-== Resolving Conflicts on the Client
+. Implement the resolver. Every conflict can be handled using a set of predefined steps, please see below for an example.
 
++
 [source,javascript]
 ----
-{
-  updateTask: async (obj, clientData, context, info) => {
-    // 1. Read the Object from the database. This example uses Knex https://knexjs.org/.
-    const task = await context.db('tasks').select().where('id', clientData.id).then((rows) => rows[0])
-
-    // 2. Conflict Detection using `VersionedObjectState`
-    //    If the version number from the database does not match the one sent by the client
-    //    A conflict occurs
-    if (conflictHandler.hasConflict(task, clientData)) {
-      // If a conflict is detected, choose to resolve on the client.
-      const { response } = conflictHandler.resolveOnClient(task, clientData)
-      return response
-    }
-
-    // 3. Always call nextState before persisting the updated record.
-    // This ensures it has the correct version number
-    conflictHandler.nextState(clientData)
-
-    // 4. Persist the update to the database
-    const update = await context.db('tasks')
-      .update(clientData)
-      .where('id', clientData.id)
-      .returning('*')
-      .then((rows) => rows[0])
-
-    return update
-  }
+// 1. Read data from data source
+const serverData = db.find(clientData.id)
+// 2. Check for conflicts
+const conflict = conflictHandler.checkForConflicts(serverData, clientData)
+// 3. If there is a conflict, return the details to the client
+if(conflict) {
+    throw conflict;
 }
+// 4. Save object to data source
+db.save(clientData.id, clientData)
 ----
 
-In the example above, `conflictHandler.resolveOnClient` is used when a conflict is detected. `resolveOnClient` returns a `response` object which should be returned to the client. The `response` contains the conflicting data and some metadata which the client can use to resolve the conflict.
+In the example above, the `throw` statement ensures the client receives all necessary data to resolve the conflict client-side. For more information about this data please see xref:error-structure[Structure of the Conflict Error].
```

Comment:
In the example above, the `throw` statement ensures the client receives all the necessary data to resolve the conflict client-side. For more information about this data, please see xref:error-structure[Structure of the Conflict Error].